### PR TITLE
Update readme in perception, code cleanup for some files.

### PR DESCRIPTION
**Pull request by lawchekun**
Update `readme.md` in `perception` folder. 
Code cleanup for some files which I did skim through. Mostly adding docstrings or removing unused imports.

**Note**: I have an code formatter in my editor which re-formats the code according to `Black`, hence most of the code changes are due to the Code formatter. There has been little change to the functional logic of most of the code.

For more info, see.
https://packagecontrol.io/packages/sublack

### CODE REVIEW

File: perception/tasks/path_marker/path_marker_detection.py [hunk 1]
```diff
@@ -204,17 +238,19 @@ def draw_marker_angles(frame, marker_angles, right=False):
...
         print('new heading:', new_heading)
 
         if new_heading is None:
             num_fails += 1
         else:
             num_fails = 0
 
-        # k = cv2.waitKey(60) & 0xff
-        if k == 27: # esc
+        k = cv2.waitKey(60) & 0xFF
```
Comment (lawchekun) [inline perception/tasks/path_marker/path_marker_detection.py:251]: Variable `k` is not defined previously.

File: perception/misc/combinedFilTest.py [hunk 2]
```diff
@@ -1,7 +1,6 @@
 import cv2
 import numpy as np
 
-import sys
```
Comment (lawchekun) [inline perception/misc/combinedFilTest.py:4]: Remove unused import

File: perception/tasks/path_marker/path_marker_detection.py [hunk 3]
```diff
@@ -9,11 +10,26 @@
     # cap = cv2.VideoCapture('../data/course_footage/path_marker_GOPR1142.mp4')
     cap = cv2.VideoCapture(args[1])
 
-def thresh_by_contour_size(frame, num_contours):
-    """ Assumes frame is grayscale """
+
+def thresh_by_contour_size(
```
Comment (lawchekun) [inline perception/tasks/path_marker/path_marker_detection.py:14]: Add docstrings for function.

File: perception/vis/TestTasks/GateSegmentationAlgo.py [hunk 4]
```diff
@@ -12,101 +13,110 @@
...
-		"""Takes in the background removed image and returns the center between
+    def analyze(self, frame: np.ndarray, debug: bool) -> Tuple[float, float]:
+        """Takes in the background removed image and returns the center between
 		the two gate posts.
 		Args:
-			frame: The background removed frame to analyze
-			debug: Whether or not tot display intermediate images for debugging
-		Reurns:
+			frame: The background removed frame to analyze.
+			debug: Whether or not to display intermediate images for debugging.
```
Comment (lawchekun) [inline perception/vis/TestTasks/GateSegmentationAlgo.py:33]: Fixed typos.

File: perception/vis/vis.py [hunk 5]
```diff
@@ -2,20 +2,20 @@
 from pathlib import Path
 from FrameWrapper import FrameWrapper
 import cv2 as cv
-import sys
-import importlib.util
-import importlib
```
Comment (lawchekun) [inline perception/vis/vis.py:7]: Remove unused imports.

File: perception/vis/TestTasks/GateSegmentationAlgo.py [hunk 6]
```diff
@@ -12,101 +13,110 @@
...
+		Returns:
```
Comment (lawchekun) [inline perception/vis/TestTasks/GateSegmentationAlgo.py:34]: Fixed typos.

**Review by Axquaris — APPROVED** [review]

**Pull request merged**